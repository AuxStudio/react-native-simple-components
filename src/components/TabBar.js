import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

import Icon from '../assets/icons/index';
import styleConstants from '../assets/styleConstants';

import Touchable from './Touchable';

const styles = StyleSheet.create({
    container: {
        ...styleConstants.regularShadow,
        height: 56,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        backgroundColor: styleConstants.primary,
    },
    tabContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        alignSelf: 'stretch',
    },
    iconContainer: {
        paddingHorizontal: 16,
        position: 'relative',
    },
    icon: {
        fontSize: styleConstants.iconFont,
    },
    text: {
        fontSize: styleConstants.smallFont,
        marginTop: 2,
    },
    countContainer: {
        position: 'absolute',
        top: -5,
        right: -4,
    },
    countText: {
        fontSize: styleConstants.smallFont,
        color: styleConstants.lightGrey,
    },
    disabled: {
        opacity: 0.33,
    },
    highlight: {
        position: 'absolute',
        top: 0,
        right: 0,
        width: 10,
        height: 10,
        borderRadius: 5,
        backgroundColor: styleConstants.secondary,
    },
});

export default (TabBar = props => {
    /* 
        PROPTYPES

        tabs: PropTypes.array.isRequired,
        backgroundColor: PropTypes.string,
        color: PropTypes.string,
        highlightedProfileTab: PropTypes.bool,

    */

    const tabs = props.tabs.map(value => {
        const colorStyles = {
            color: value.active
                ? styleConstants.secondary
                : props.color ? props.color : styleConstants.white,
        };

        const count = value.count ? (
            <View style={styles.countContainer}>
                <Text style={[styles.countText, styleConstants.primaryFont]}>
                    {value.count.toString()}
                </Text>
            </View>
        ) : null;

        const highlight = props.highlightProfileTab &&
        value.title === 'Profile' &&
        !value.active && <View style={styles.highlight} />;

        const tab = value.disabled ? (
            <View
                style={[styles.tabContainer, styles.disabled]}
                key={'tab-' + value.title}>
                <View style={styles.iconContainer}>
                    <Icon
                        name={value.icon}
                        style={[styles.icon, colorStyles]}
                    />
                    {count}
                </View>
                <Text
                    style={[
                        styles.text,
                        styleConstants.primaryFont,
                        colorStyles,
                    ]}>
                    {value.title}
                </Text>
            </View>
        ) : (
            <Touchable
                onPress={value.action}
                style={styles.tabContainer}
                key={'tab-' + value.title}>
                <View style={styles.iconContainer}>
                    <Icon
                        name={value.icon}
                        style={[styles.icon, colorStyles]}
                    />
                    {count}
                    {highlight}
                </View>
                <Text
                    style={[
                        styles.text,
                        styleConstants.primaryFont,
                        colorStyles,
                    ]}>
                    {value.title}
                </Text>
            </Touchable>
        );

        return tab;
    });

    const backgroundColorStyles = props.backgroundColor
        ? {
              backgroundColor: props.backgroundColor,
          }
        : null;

    return (
        <View style={[styles.container, backgroundColorStyles]}>{tabs}</View>
    );
});
