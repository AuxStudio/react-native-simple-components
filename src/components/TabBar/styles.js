import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const HEIGHT = 56;

const styles = StyleSheet.create({
  scrollWrapper: {
    maxHeight: HEIGHT,
  },
  scrollContainer: {
    height: HEIGHT,
  },
  container: {
    height: HEIGHT,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: styleConstants.colors.white,
  },
  tabContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
  },
  iconContainer: {
    paddingHorizontal: styleConstants.dimensions.padding.large,
    position: 'relative',
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.secondaryText,
  },
  text: {
    fontSize: styleConstants.fonts.sizes.small,
    color: styleConstants.colors.secondaryText,
    textAlign: 'center',
  },
  disabled: {
    opacity: 0.33,
  },
});

export default styles;
