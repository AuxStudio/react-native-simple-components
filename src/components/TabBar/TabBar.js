import React from 'react';
import PropTypes from 'prop-types';
import { View, ScrollView, Text, ViewPropTypes } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';
import styleConstants from '../../styleConstants';
import Touchable from '../Touchable';

const propTypes = {
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      iconName: PropTypes.string,
      customIcon: PropTypes.node,
      disabled: PropTypes.bool,
    }),
  ).isRequired,
  activeTab: PropTypes.string,
  handlePress: PropTypes.func,
  androidRipple: PropTypes.bool,
  androidRippleColor: PropTypes.string,
  androidRippleBorderless: PropTypes.bool,
  shouldScrollHorizontally: PropTypes.bool,
  tabWidth: PropTypes.number, // works with shouldScrollHorizontally
  showShadow: PropTypes.bool,
  textColor: PropTypes.string, // both icons and text
  activeTextColor: PropTypes.string, // both icons and text
  iconStyle: Text.propTypes.style,
  textStyle: Text.propTypes.style,
  tabStyle: ViewPropTypes.style,
  activeTabStyle: ViewPropTypes.style,
  disabledTabStyle: ViewPropTypes.style,
  style: ViewPropTypes.style,
  wrapperStyle: ViewPropTypes.style, // works with shouldScrollHorizonally
};

const defaultProps = {
  textColor: styleConstants.colors.primaryText,
  activeTextColor: styleConstants.colors.primary,
};

const TabBar = ({
  tabs,
  activeTab,
  handlePress,
  androidRipple,
  androidRippleColor,
  androidRippleBorderless,
  shouldScrollHorizontally,
  tabWidth,
  showShadow,
  textColor,
  activeTextColor,
  iconStyle,
  textStyle,
  tabStyle,
  activeTabStyle,
  disabledTabStyle,
  style,
  wrapperStyle,
}) => {
  const shadowStyles = showShadow && styleConstants.shadows.regular;

  const tabBarComponent =
    tabs &&
    tabs.map((tab) => {
      const color = activeTab === tab.title ? activeTextColor : textColor;
      const colorStyles = {
        color,
      };
      const activeTabStyles = activeTab === tab.title && activeTabStyle;
      const disabledStyles = [];

      if (tab.disabled) {
        disabledStyles.push(styles.disabled);
        // Add the custom disabled style if provided
        if (disabledTabStyle) {
          disabledStyles.push(disabledTabStyle);
        }
      }

      const iconComponent = tab.customIcon
        ? tab.customIcon
        : tab.iconName && (
            <Icon name={tab.iconName} style={[styles.icon, colorStyles, iconStyle]} />
          );

      const titleTextComponent = tab.title && (
        <Text style={[styles.text, colorStyles, textStyle]}>{tab.title}</Text>
      );

      return (
        <Touchable
          key={tab.title}
          onPress={() => handlePress && handlePress(tab.title)}
          androidRipple={androidRipple}
          androidRippleColor={androidRippleColor}
          androidRippleBorderless={androidRippleBorderless}
          disabled={tab.disabled}
          style={[
            styles.tabContainer,
            {
              width: tabWidth,
            },
            activeTabStyles,
            disabledStyles,
            tabStyle,
          ]}
        >
          {iconComponent}
          {titleTextComponent}
        </Touchable>
      );
    });

  const tabBarWrapperComponent = shouldScrollHorizontally ? (
    <ScrollView
      horizontal
      showsHorizontalScrollIndicator={false}
      style={[styles.scrollWrapper, shadowStyles, wrapperStyle]}
      contentContainerStyle={[styles.scrollContainer, style]}
    >
      {tabBarComponent}
    </ScrollView>
  ) : (
    <View style={[styles.container, shadowStyles, style]}>{tabBarComponent}</View>
  );

  return tabBarWrapperComponent;
};

TabBar.propTypes = propTypes;
TabBar.defaultProps = defaultProps;

export default TabBar;
