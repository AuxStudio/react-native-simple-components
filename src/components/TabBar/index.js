import React from 'react';
import PropTypes from 'prop-types';
import { View, ScrollView, Text, ViewPropTypes } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';
import styleConstants from '../../styleConstants';
import Touchable from '../Touchable';

const propTypes = {
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      iconName: PropTypes.string,
      customIcon: PropTypes.node,
      disabled: PropTypes.bool,
    }),
  ).isRequired,
  activeTab: PropTypes.string, // active tab title
  activeTabIndex: PropTypes.number, // or if you'd prefer to use tab index
  handlePress: PropTypes.func,
  androidRipple: PropTypes.bool,
  androidRippleColor: PropTypes.string,
  androidRippleBorderless: PropTypes.bool,
  shouldScrollHorizontally: PropTypes.bool,
  tabWidth: PropTypes.number, // works with shouldScrollHorizontally
  showShadow: PropTypes.bool,
  textColor: PropTypes.string, // both icons and text
  activeTextColor: PropTypes.string, // both icons and text
  iconStyle: Text.propTypes.style,
  textStyle: Text.propTypes.style,
  tabStyle: ViewPropTypes.style,
  activeTabStyle: ViewPropTypes.style,
  disabledTabStyle: ViewPropTypes.style,
  style: ViewPropTypes.style,
  wrapperStyle: ViewPropTypes.style, // works with shouldScrollHorizonally
  testID: PropTypes.string,

  // used to create the individual tab testIDs, useful if more than one TabBar is present
  tabTestIDPredix: PropTypes.string,
};

const defaultProps = {
  textColor: styleConstants.colors.primaryText,
  activeTextColor: styleConstants.colors.primary,
  tabTestIDPredix: 'tabBar.tab',
};

const TabBar = ({
  tabs,
  activeTab,
  activeTabIndex,
  handlePress,
  androidRipple,
  androidRippleColor,
  androidRippleBorderless,
  shouldScrollHorizontally,
  tabWidth,
  showShadow,
  textColor,
  activeTextColor,
  iconStyle,
  textStyle,
  tabStyle,
  activeTabStyle,
  disabledTabStyle,
  style,
  wrapperStyle,
  testID,
  tabTestIDPredix,
}) => {
  const shadowStyles = showShadow && styleConstants.shadows.regular;

  const tabBarComponent =
    tabs &&
    tabs.map((tab, index) => {
      const isActive = activeTab === tab.title || activeTabIndex === index;

      const color = isActive ? activeTextColor : textColor;
      const colorStyles = {
        color,
      };
      const activeTabStyles = isActive && activeTabStyle;
      const disabledStyles = [];

      if (tab.disabled) {
        disabledStyles.push(styles.disabled);
        // Add the custom disabled style if provided
        if (disabledTabStyle) {
          disabledStyles.push(disabledTabStyle);
        }
      }

      const iconComponent = tab.customIcon
        ? tab.customIcon
        : tab.iconName && (
            <Icon name={tab.iconName} style={[styles.icon, colorStyles, iconStyle]} />
          );

      const titleTextComponent = tab.title && (
        <Text style={[styles.text, colorStyles, textStyle]}>{tab.title}</Text>
      );

      return (
        <Touchable
          key={tab.title}
          onPress={() => handlePress && handlePress(tab)}
          androidRipple={androidRipple}
          androidRippleColor={androidRippleColor}
          androidRippleBorderless={androidRippleBorderless}
          disabled={tab.disabled}
          style={[
            styles.tabContainer,
            {
              width: tabWidth,
            },
            activeTabStyles,
            disabledStyles,
            tabStyle,
          ]}
          testID={`${tabTestIDPredix}.${tab.title}`}
        >
          {iconComponent}
          {titleTextComponent}
        </Touchable>
      );
    });

  const tabBarWrapperComponent = shouldScrollHorizontally ? (
    <ScrollView
      horizontal
      showsHorizontalScrollIndicator={false}
      style={[styles.scrollWrapper, shadowStyles, wrapperStyle]}
      contentContainerStyle={[styles.scrollContainer, style]}
      testID={testID}
    >
      {tabBarComponent}
    </ScrollView>
  ) : (
    <View style={[styles.container, shadowStyles, style]}>{tabBarComponent}</View>
  );

  return tabBarWrapperComponent;
};

TabBar.propTypes = propTypes;
TabBar.defaultProps = defaultProps;

export default TabBar;
