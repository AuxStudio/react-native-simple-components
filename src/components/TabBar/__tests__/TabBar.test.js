/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

import TabBar from '../';

const TABS = [
  {
    title: 'Home',
    iconName: 'home',
  },
  {
    title: 'Profile',
    iconName: 'person',
    disabled: true,
  },
];
const ACTIVE_TAB = 'Home';

it('renders a TabBar', () => {
  expect(
    renderer.create(
      <TabBar
        tabs={TABS}
        activeTab={ACTIVE_TAB}
        handlePress={jest.fn()}
        androidRipple
        androidRippleColor="red"
        androidRippleBorderless
        showShadow
        textColor="red"
        activeTextColor="blue"
        iconStyle={{ color: 'red' }}
        textStyle={{ color: 'red' }}
        tabStyle={{ backgroundColor: 'blue' }}
        activeTabStyle={{ backgroundColor: 'green' }}
        disabledTabStyle={{ backgroundColor: 'purple' }}
        style={{ backgroundColor: 'orange' }}
        wrapperStyle={{ backgroundColor: 'pink' }}
      />,
    ),
  ).toMatchSnapshot();
});

it('renders a TabBar with customIcon', () => {
  expect(
    renderer.create(
      <TabBar
        tabs={[
          {
            title: 'Home',
            customIcon: <MaterialIcon name="check" />,
          },
        ]}
        activeTab={ACTIVE_TAB}
      />,
    ),
  ).toMatchSnapshot();
});

it('renders a TabBar', () => {
  expect(
    renderer.create(
      <TabBar tabs={TABS} activeTab={ACTIVE_TAB} shouldScrollHorizontally tabWidth={200} />,
    ),
  ).toMatchSnapshot();
});
/* eslint-enable */
