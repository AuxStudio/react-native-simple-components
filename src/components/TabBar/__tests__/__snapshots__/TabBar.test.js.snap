// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar renders horizontally 1`] = `
<RCTScrollView
  contentContainerStyle={
    Array [
      Object {
        "height": 56,
      },
      undefined,
    ]
  }
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Array [
      Object {
        "maxHeight": 56,
      },
      undefined,
      undefined,
    ]
  }
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": 200,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#757575",
                "fontSize": 24,
              },
              Object {
                "color": "#3D84A8",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#757575",
              "fontSize": 15,
              "textAlign": "center",
            },
            Object {
              "color": "#3D84A8",
            },
            undefined,
          ]
        }
      >
        Home
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0.33,
          "width": 200,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#757575",
                "fontSize": 24,
              },
              Object {
                "color": "#212121",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#757575",
              "fontSize": 15,
              "textAlign": "center",
            },
            Object {
              "color": "#212121",
            },
            undefined,
          ]
        }
      >
        Profile
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`TabBar renders with all props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
      },
      Object {
        "borderColor": null,
        "borderWidth": 0,
        "elevation": 6,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 2,
      },
      Object {
        "backgroundColor": "orange",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "blue",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "width": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#757575",
              "fontSize": 24,
            },
            Object {
              "color": "blue",
            },
            Object {
              "color": "red",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#757575",
            "fontSize": 15,
            "textAlign": "center",
          },
          Object {
            "color": "blue",
          },
          Object {
            "color": "red",
          },
        ]
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "blue",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0.33,
        "width": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#757575",
              "fontSize": 24,
            },
            Object {
              "color": "red",
            },
            Object {
              "color": "red",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#757575",
            "fontSize": 15,
            "textAlign": "center",
          },
          Object {
            "color": "red",
          },
          Object {
            "color": "red",
          },
        ]
      }
    >
      Profile
    </Text>
  </View>
</View>
`;

exports[`TabBar renders with customIcon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "width": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#757575",
            "fontSize": 15,
            "textAlign": "center",
          },
          Object {
            "color": "#3D84A8",
          },
          undefined,
        ]
      }
    >
      Home
    </Text>
  </View>
</View>
`;

exports[`TabBar renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "width": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#757575",
              "fontSize": 24,
            },
            Object {
              "color": "#3D84A8",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#757575",
            "fontSize": 15,
            "textAlign": "center",
          },
          Object {
            "color": "#3D84A8",
          },
          undefined,
        ]
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0.33,
        "width": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#757575",
              "fontSize": 24,
            },
            Object {
              "color": "#212121",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#757575",
            "fontSize": 15,
            "textAlign": "center",
          },
          Object {
            "color": "#212121",
          },
          undefined,
        ]
      }
    >
      Profile
    </Text>
  </View>
</View>
`;
